lib/read/cardano-wallet-read.cabal

```
@@ -44,99 +41,132 @@ flag release
+  exposed-modules:
+    Cardano.Wallet.Read
+    Cardano.Wallet.Read.Block
+    Cardano.Wallet.Read.Block.BlockNo
+    Cardano.Wallet.Read.Block.Gen
+    Cardano.Wallet.Read.Block.Gen.Babbage
+    Cardano.Wallet.Read.Block.Gen.BlockParameters
+    Cardano.Wallet.Read.Block.Gen.Build
+    Cardano.Wallet.Read.Block.Gen.Byron
+    Cardano.Wallet.Read.Block.Gen.Shelley
+    Cardano.Wallet.Read.Block.HeaderHash
+    Cardano.Wallet.Read.Block.SlotNo
+    Cardano.Wallet.Read.Eras
+    Cardano.Wallet.Read.Eras.EraFun
+    Cardano.Wallet.Read.Eras.EraValue
+    Cardano.Wallet.Read.Eras.InAnyCardanoEra
+    Cardano.Wallet.Read.Eras.KnownEras
+    Cardano.Wallet.Read.Tx
+    Cardano.Wallet.Read.Tx.Cardano
+    Cardano.Wallet.Read.Tx.CBOR
+    Cardano.Wallet.Read.Tx.Certificates
+    Cardano.Wallet.Read.Tx.CollateralInputs
+    Cardano.Wallet.Read.Tx.CollateralOutputs
+    Cardano.Wallet.Read.Tx.Eras
+    Cardano.Wallet.Read.Tx.ExtraSigs
+    Cardano.Wallet.Read.Tx.Fee
+    Cardano.Wallet.Read.Tx.Gen
+    Cardano.Wallet.Read.Tx.Gen.Address
+    Cardano.Wallet.Read.Tx.Gen.Allegra
+    Cardano.Wallet.Read.Tx.Gen.Alonzo
+    Cardano.Wallet.Read.Tx.Gen.Babbage
+    Cardano.Wallet.Read.Tx.Gen.Byron
+    Cardano.Wallet.Read.Tx.Gen.Conway
+    Cardano.Wallet.Read.Tx.Gen.Mary
+    Cardano.Wallet.Read.Tx.Gen.Shelley
+    Cardano.Wallet.Read.Tx.Gen.TxParameters
+    Cardano.Wallet.Read.Tx.Hash
+    Cardano.Wallet.Read.Tx.Inputs
+    Cardano.Wallet.Read.Tx.Integrity
+    Cardano.Wallet.Read.Tx.Metadata
+    Cardano.Wallet.Read.Tx.Mint
+    Cardano.Wallet.Read.Tx.Outputs
+    Cardano.Wallet.Read.Tx.ReferenceInputs
+    Cardano.Wallet.Read.Tx.ScriptValidity
+    Cardano.Wallet.Read.Tx.Validity
+    Cardano.Wallet.Read.Tx.Withdrawals
+    Cardano.Wallet.Read.Tx.Witnesses
 
-    hs-source-dirs:   lib
-    default-language: Haskell2010
+  build-depends:
+    , base
+    , bytestring
+    , cardano-api
+    , cardano-crypto
+    , cardano-crypto-class
+    , cardano-crypto-praos
+    , cardano-crypto-test
+    , cardano-crypto-wrapper
+    , cardano-ledger-allegra
+    , cardano-ledger-alonzo
+    , cardano-ledger-alonzo-test
+    , cardano-ledger-api
+    , cardano-ledger-babbage
+    , cardano-ledger-babbage-test
+    , cardano-ledger-binary
+    , cardano-ledger-byron
+    , cardano-ledger-byron-test
+    , cardano-ledger-conway
+    , cardano-ledger-core
+    , cardano-ledger-mary
+    , cardano-ledger-shelley
+    , cardano-ledger-shelley-ma-test
+    , cardano-ledger-shelley-test
+    , cardano-protocol-tpraos
+    , cardano-strict-containers
+    , containers
+    , data-array-byte
+    , deepseq
+    , directory
+    , extra
+    , fmt
+    , generic-lens
+    , generics-sop
+    , hedgehog
+    , lens
+    , memory
+    , mtl
+    , operational
+    , ouroboros-consensus
+    , ouroboros-consensus-cardano
+    , ouroboros-consensus-protocol
+    , ouroboros-network-api
+    , QuickCheck
+    , small-steps
+    , small-steps-test
+    , text
+    , text-class
+
```

```suggestion
    , transformers
```

In order to keep the extension footprint low it is generally discouraged to depend on `mtl` when `transformers` will do. Here, we only need `transformers`.

`Endo` is in `base`, `lift` in `Control.Monad.Trans.Class`.